src/js/components/generateFiles/GenerateFilesContent.jsx

```diff
@@ -95,50 +122,73 @@ export default class GenerateFilesContent extends React.Component {
-                                <button
-                                    id="agency-toggle__info-icon"
-                                    className="agency-toggle__info-icon"
-                                    onFocus={this.showTooltip}
-                                    onBlur={this.closeTooltip}
-                                    onMouseLeave={this.closeTooltip}
-                                    onMouseEnter={this.showTooltip} >
-                                    <InfoCircle />
-                                </button>
-                            </div>
-                        </span>
-                    </div>
-
                     <div className="usa-da-generate-content">
+                        <div className="agency-toggle">
+                            <div className="agency-toggle__text">
+                                Generate File D1 from records where my agency is the:
+                            </div>
+                            <AgencyToggle
+                                funding={d1.fundingAgency}
+                                toggleAgencyType={this.toggleAgencyType.bind(this, "d1")} />
+                            <span className="agency-toggle__info-icon-holder">
+                                <div ref={(div) => {
+                                    this.d1ReferenceDiv = div;
+                                }}>
+                                    {d1Tooltip}
+                                    <button
+                                        id="agency-toggle__info-icon"
+                                        className="agency-toggle__info-icon"
+                                        onFocus={this.showTooltip.bind(this, 'd1')}
+                                        onBlur={this.closeTooltip.bind(this, 'd1')}
+                                        onMouseLeave={this.closeTooltip.bind(this, 'd1')}
+                                        onMouseEnter={this.showTooltip.bind(this, 'd1')} >
+                                        <InfoCircle />
+                                    </button>
+                                </div>
+                            </span>
+                        </div>
                         <GenerateFileBox
                             label="File D1: Procurement Awards (FPDS data)"
                             datePlaceholder="Sign"
                             startingTab={1}
-                            value={this.props.d1}
-                            error={this.props.d1.error}
-                            showDownload={this.props.d1.showDownload}
+                            value={d1}
+                            error={d1.error}
+                            showDownload={d1.showDownload}
                             onDateChange={this.handleDateChange.bind(this, "d1")}
                             updateError={this.updateError.bind(this, "d1")}
                             clickedDownload={this.clickedDownload.bind(this, "D1")} />
 
```

Comment:
General tech debt here -- putting `.bind()` in the render creates a new anonymous fn each render. This is everywhere in broker and IMO should be put in the constructor for performance reasons:

https://reactjs.org/docs/faq-functions.html

Reply:
Thank you. I am aware binding in the render is a horrible practice. Also, I am a believer of not piling crap on top of crap.  I am also not a fan of binding in the constructor or binding when it is not necessary at all. However, my first attempt at this PR included pulling in more babel presets allowing us to write arrow functions in our classes keeping `this` which would fix both binding in render and binding at all, was biting off more than I could chew at this moment due to the fact that we are continuously wrapping props in other methods and then binding again further down the tree. Due to the time constraint of this sprint, I took a step back and instead of refactoring multiple components, I decided to complete the task of this ticket. I do believe we need to have a discussion of refactoring not only the binds but a lot of this web app.